File: MediATech/Model/BL/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MediATech.Model.BL
{
    public class Book : MediaObject
    {
        private string _isbn;
        private string _author;
        private string _publisher;
        private int _pageCount;
        

        public Book(int id, string title, bool available, string state, DateOnly publicationDate, string isbn, string author, string publisher, int pageCount)
            : base(id, title, available, state, publicationDate)
        {
            ISBN = isbn;
            Author = author;
            Publisher = publisher;
            PageCount = pageCount;
        }


        public string ISBN { get => _isbn; set => _isbn = value; }

        public string Author { get =>  _author; set => _author = value;}

        public string Publisher { get => _publisher; set => _publisher = value;}

        public int PageCount { get => _pageCount; set => _pageCount = value;}
    }
}

File: src/main/java/mediatech/Model/BL/Book.java
package mediatech.Model.BL;

import java.util.Date;

public class Book extends MediaObject {

    private String isbn;
    private String author;
    private String publisher;
    private int pageCount;

    public Book(int id, String title, boolean available, String state, Date publicationDate, String isbn, String author, String publisher, int pageCount) {
        super(id, title, available, state, publicationDate);

        this.isbn = isbn;
        this.author = author;
        this.publisher = publisher;
        this.pageCount = pageCount;
    }

    public Book() {}

    public String getIsbn() {
        return isbn;
    }
    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getAuthor() {
        return author;
    }
    public void setAuthor(String author) {
        this.author = author;
    }   

    public String getPublisher() {
        return publisher;
    }    
    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }   
     
    public int getPageCount() {
        return pageCount;
    }
    public void setPageCount(int pageCount) {
        this.pageCount = pageCount;
    }
    
}
